### PR TITLE
UefiDevicePathLib: remove PEIM support

### LINKED ISSUE #306: [Bug]: UefiDevicePathLib Should Not Support PEIMs
### Is there an existing issue for this?

- [X] I have searched existing issues

### Current Behavior

[MdePkg/Library/UefiDevicePathLib/UefiDevicePathLib.inf](https://github.com/microsoft/mu_basecore/blob/release/202208/MdePkg/Library/UefiDevicePathLib/UefiDevicePathLib.inf) had `PEIM` added to the `LIBRARY_CLASS` list as of commit acf286f.

This library class instance includes code such as that in `DevicePathUtilitiesDxeSmm.c` which directly links against `gBS` which is provided by `UefiBootServicesTableLib` and not available in PEI.

### Expected Behavior

UefiDevicePathLib should be refactored for `PEIM` support or drop `PEIM` from the support library class types.

### Steps To Reproduce

Review the module.

### Build Environment

```markdown
- OS(s): All
- Tool Chain(s): All
- Targets Impacted: All
```

### Version Information

```text
Commit: acf286f
```

### Urgency

Low

### Are you going to fix this?

Someone else needs to fix it

### Do you need maintainer feedback?

No maintainer feedback needed

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by Javagedes**
## Description

Removed PEIM from the list of supported modules for UefiDevicePathLib.

UefiDevicePathLib incorrectly described support for PEIM even though the implementation for DevicePathFromHandle() uses gBS which is not available in PEIM.

closes #306 

For each item, place an "x" in between `[` and `]` if true. Example: `[x]`.
_(you can also check items in the GitHub UI)_

- [ ] Impacts functionality?
  - **Functionality** - Does the change ultimately impact how firmware functions?
  - Examples: Add a new library, publish a new PPI, update an algorithm, ...
- [ ] Impacts security?
  - **Security** - Does the change have a direct security impact on an application,
    flow, or firmware?
  - Examples: Crypto algorithm change, buffer overflow fix, parameter
    validation improvement, ...
- [x] Breaking change?
  - **Breaking change** - Will anyone consuming this change experience a break
    in build or boot behavior?
  - Examples: Add a new library class, move a module to a different repo, call
    a function in a new library class in a pre-existing module, ...
- [ ] Includes tests?
  - **Tests** - Does the change include any explicit test code?
  - Examples: Unit tests, integration tests, robot tests, ...
- [ ] Includes documentation?
  - **Documentation** - Does the change contain explicit documentation additions
    outside direct code modifications (and comments)?
  - Examples: Update readme file, add feature readme file, link to documentation
    on an a separate Web page, ...

## How This Was Tested

Confirmed through CI

## Integration Instructions

PEIM Modules consuming `UefiDevicePathLib` instance of `DevicePathLib` must use `UefiDevicePathLibBase`

### CODE REVIEW

File: MdeModulePkg/MdeModulePkg.dsc [hunk 1]
```diff
@@ -158,15 +158,12 @@
 ## MU_CHANGE - SmmLockBoxPeiLib only supports IA32 and X64.
   LockBoxLib|MdeModulePkg/Library/LockBoxNullLib/LockBoxNullLib.inf
   PcdDatabaseLoaderLib|MdeModulePkg/Library/PcdDatabaseLoaderLib/Pei/PcdDatabaseLoaderLibPei.inf  # MU_CHANGE
+  DevicePathLib|MdePkg/Library/UefiDevicePathLib/UefiDevicePathLibBase.inf  # MU_CHANGE
```
Comment (TaylorBeebe) [inline MdeModulePkg/MdeModulePkg.dsc]: I think if you move this right under LockBoxLib we can drop the MU_CHANGE tag

File: MdePkg/Library/UefiDevicePathLib/UefiDevicePathLib.inf [hunk 2]
```diff
@@ -17,7 +17,7 @@
   FILE_GUID                      = 91c1677a-e57f-4191-8b8e-eb7711a716e0
   MODULE_TYPE                    = UEFI_DRIVER
   VERSION_STRING                 = 1.0
-  LIBRARY_CLASS                  = DevicePathLib|PEIM DXE_CORE DXE_DRIVER DXE_RUNTIME_DRIVER DXE_SMM_DRIVER UEFI_APPLICATION UEFI_DRIVER SMM_CORE MM_STANDALONE
+  LIBRARY_CLASS                  = DevicePathLib|DXE_CORE DXE_DRIVER DXE_RUNTIME_DRIVER DXE_SMM_DRIVER UEFI_APPLICATION UEFI_DRIVER SMM_CORE MM_STANDALONE  # MU_CHANGE
```
Comment (TaylorBeebe) [inline MdePkg/Library/UefiDevicePathLib/UefiDevicePathLib.inf]: I think we can drop the MU_CHANGE tag here

**Comment by TaylorBeebe** [comment]
It might be helpful to whoever does the next integration to call out that this PR reverts https://github.com/microsoft/mu_basecore/commit/4debc5d11a8f2ab84793aece93813b4d1a771ca0

**Review by TaylorBeebe — APPROVED** [review]

File: MdePkg/Library/UefiDevicePathLib/UefiDevicePathLib.inf (hunk 2 above)
Comment (Javagedes) [reply MdePkg/Library/UefiDevicePathLib/UefiDevicePathLib.inf]: Updated. Thanks!

File: MdeModulePkg/MdeModulePkg.dsc (hunk 1 above)
Comment (Javagedes) [reply MdeModulePkg/MdeModulePkg.dsc]: Moved, Thanks!

**Review by makubacki — APPROVED** [review]

**Pull request merged**